### PR TITLE
Don't add ActionDispatch::Static middleware continued

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?

I am attempting to finish #1091 by adding prism through a cdn when no asset pipeline is present

### What approach did you choose and why?

When their is no asset pipeline I decided to serve prism through a cdn

### CODE REVIEW

File: app/helpers/preview_helper.rb [hunk 1]
```diff
@@ -2,14 +2,22 @@
 
 module PreviewHelper
   AVAILABLE_PRISM_LANGUAGES = %w[ruby erb haml]
-  FALLBACK_LANGUAGE = "ruby"
+  FALLBACK_LANGUAGE = "ruby"\
```
Comment (camertron) [inline app/helpers/preview_helper.rb:5]:
```suggestion
  FALLBACK_LANGUAGE = "ruby"
```

File: app/helpers/preview_helper.rb [hunk 2]
```diff
@@ -2,14 +2,22 @@
...
 
   def preview_source
     return if @render_args.nil?
 
     render "preview_source"
   end
 
+  def prism_css_source
```
Comment (camertron) [inline app/helpers/preview_helper.rb:13]:
```suggestion
  def prism_css_source_url
```

File: app/helpers/preview_helper.rb [hunk 3]
```diff
@@ -2,14 +2,22 @@
...
+    serve_static_previews? ? asset_path('prism.css', skip_pipeline: true) : 'https://cdn.jsdelivr.net/npm/prismjs@1.28.0/themes/prism.min.css'
+  end
+
+  def prism_js_source
```
Comment (camertron) [inline app/helpers/preview_helper.rb:17]:
```suggestion
  def prism_js_source_url
```

File: app/helpers/preview_helper.rb [hunk 4]
```diff
@@ -62,4 +70,8 @@ def prism_language_name_by_template_path(template_file_path:)
 
     language
   end
+
+  def serve_static_previews?
```
Comment (camertron) [inline app/helpers/preview_helper.rb:74]:
```suggestion
  def serve_static_preview_assets?
```

File: lib/view_component/engine.rb [hunk 5]
```diff
@@ -110,11 +110,15 @@ class Engine < Rails::Engine # :nodoc:
     end
 
     initializer "static assets" do |app|
-      if app.config.view_component.show_previews
+      if serve_static_previews? app.config
         app.middleware.use(::ActionDispatch::Static, "#{root}/app/assets/vendor")
       end
     end
 
+    def serve_static_previews?(app_config)
```
Comment (camertron) [inline lib/view_component/engine.rb:118]: Can this be extracted into a module perhaps?

```suggestion
    def serve_static_preview_assets?(app_config)
```

**Review by camertron — CHANGES_REQUESTED** [review]
Thanks for picking this up @reeganviljoen!

**Comment by reeganviljoen** [comment]
@camertron can you please re-review this when you can :pray:

**Review by camertron — APPROVED** [review]
Thanks @reeganviljoen, looks good 👍

**Comment by reeganviljoen** [comment]
@camertron Its been rebased and is ready to merge 🙏

**Review by Spone — CHANGES_REQUESTED** [review]

**Review by camertron — APPROVED** [review]

**Pull request merged**